File: Lab9/Q1.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, i, j, k;
    char temp[50];  // Assuming each name has a maximum length of 50 characters
    printf("Enter the number of names to be sorted: ");
    scanf("%d", &n);

    char names[n][50];  // Array to store names

    for (i = 0; i < n; i++) {
        printf("Enter name %d: ", i + 1);
        scanf("%s", names[i]);  // Read names as strings
    }

   // Insertion Sort in Descending Order
for (i = 1; i < n; i++) {
    for (j = i - 1; j >= 0; j--) {
        if (strcmp(names[i], names[j]) > 0) {  // Corrected condition for descending order
            break;
        }
    }
    if (i - j > 1) {
        strcpy(temp, names[i]);
        for (k = i; k > j + 1; k--) {
            strcpy(names[k], names[k - 1]);
        }
        strcpy(names[j + 1], temp);
    }
}


    // Sorted List Output in Descending Order
    printf("\nSorted list of names in descending order is:\n");
    for (i = n - 1; i >= 0; i--) {
        printf("%s\n", names[i]);
    }

    return 0;
}

File: Lab9/Q1.java
import java.util.Scanner;
public class Q1 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n, i, j, k;
        System.out.print("Enter number of words to be sorted: ");
        n = scanner.nextInt();
        String[] arr = new String[n];

        for (i = 0; i < n; i++) {
            System.out.print("Enter word " + (i + 1) + ": ");
            arr[i] = scanner.next();
        }

        // Insertion Sort for words
        for (i = 1; i < n; i++) {
            for (j = i - 1; j >= 0; j--) {
                if (arr[i].compareTo(arr[j]) <= 0) {
                    break;
                }
            }
            if (i - j > 1) {
                String temp = arr[i];
                for (k = i; k > j + 1; k--) {
                    arr[k] = arr[k - 1];
                }
                arr[j + 1] = temp;
            }
        }

        // Sorted List Output
        System.out.println("\nSorted list is:");
        for (i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}
